### DISCUSSION TITLE
manually refetch?

### DISCUSSION THREAD

**Discussion by chj-damon** [vercel/swr#1295] [Q&A]
```js
const { data, isValidating, error } = useSWR('xxx');
```

what should I do if I want to refetch this request manually? can we provide a method to the result?

```js
const { data, isValidating, error, refetch } = useSWR('xxx');
```

**Comment by huozhi** [comment]
It returns a bound `mutate` function from the hook.

```js
const { data, mutate } = useSWR('key')

mutate() // revalidate
```

check https://swr.vercel.app/docs/mutation#bound-mutate

**Reply by chj-damon** [reply]
you mean `mutate` is the same as `refetch` ?

**Reply by chj-damon** [reply]
`mutate` is used to modify cached data locally, `refetch` is used to refetch remote data again. I think they are different.

**Reply by huozhi** [reply]
`mutate(data, shouldRevalidate)`

the 2nd argument of the bound mutate is `shouldRevalidate` by default it `true`, which could be used for triggering another revalidation. We're deprecating the returned `revalidate` and recommend to use this one.

**Reply by chj-damon** [reply]
fair enough.
